### PR TITLE
Add processor and intergration test for qwen3vl

### PR DESCRIPTION
**Pull request by JJJYmmm**
# What does this PR do?

- support aux loss in qwen3vlmoe
- update qwen3vl processor test since the Qwen3-VL-235B-A22B is released!
- fix multi-video inference
- add intergration test for 30a3 (skip for now)

# Who can review?
- text models: @ArthurZucker @Cyrilvallez
- multimodal models: @zucchini-nlp

### CODE REVIEW

File: src/transformers/models/qwen3_vl/modular_qwen3_vl.py [hunk 1]
```diff
@@ -1007,6 +1007,7 @@ def get_video_features(
 
     @auto_docstring
     @can_return_tuple
+    @check_model_inputs
```
Comment (zucchini-nlp) [inline src/transformers/models/qwen3_vl/modular_qwen3_vl.py]: we don't need both afaik, `check_model_inputs` will handle tuple/non-tuple as well

File: src/transformers/models/qwen3_vl/modular_qwen3_vl.py [hunk 2]
```diff
@@ -1396,7 +1394,7 @@ def __call__(
             index = 0
             for i in range(len(text)):
                 while self.video_token in text[i]:
-                    metadata = video_metadata[i]
+                    metadata = video_metadata[index]
```
Comment (zucchini-nlp) [inline src/transformers/models/qwen3_vl/modular_qwen3_vl.py:1396]: also just merged a PR fixing it 😄

File: tests/models/qwen3_vl/test_processing_qwen3_vl.py [hunk 3]
```diff
@@ -299,6 +292,9 @@ def test_apply_chat_template_video_frame_sampling(self):
         out_dict = processor.apply_chat_template(messages, add_generation_prompt=True, tokenize=True, return_dict=True)
         self.assertListEqual(list(out_dict.keys()), ["input_ids", "attention_mask"])
 
+        # for fast test, set the longest edge to 4096
+        processor.video_processor.size["longest_edge"] = 8192
```
Comment (zucchini-nlp) [inline tests/models/qwen3_vl/test_processing_qwen3_vl.py]: ultra nit: the comment doesn't align with code

File: tests/models/qwen3_vl_moe/test_modeling_qwen3_vl_moe.py [hunk 4]
```diff
@@ -296,3 +301,280 @@ def test_video_forward(self):
                 video_grid_thw=video_grid_thw,
             )
             self.assertIsNotNone(outputs)
+
+
+@require_torch
+@unittest.skip("The checkpoint is not yet released")
```
Comment (zucchini-nlp) [inline tests/models/qwen3_vl_moe/test_modeling_qwen3_vl_moe.py:307]: we can remove the skip now, right?

File: tests/models/qwen3_vl_moe/test_modeling_qwen3_vl_moe.py [hunk 5]
```diff
@@ -296,3 +301,280 @@ def test_video_forward(self):
...
+    def test_small_model_integration_test(self):
+        model = Qwen3VLMoeForConditionalGeneration.from_pretrained(
+            "Qwen/Qwen3-VL-30B-A3B-Instruct", dtype="auto", device_map="auto"
+        )
+
+        inputs = self.processor.apply_chat_template(
+            self.message, tokenize=True, add_generation_prompt=True, return_dict=True, return_tensors="pt"
+        )
+        expected_input_ids = [151644, 872, 198, 151652, 151655, 151655, 151655, 151655, 151655, 151655, 151655, 151655, 151655, 151655, 151655, 151655, 151655]  # fmt: skip
+        assert expected_input_ids == inputs.input_ids[0].tolist()[:17]
```
Comment (zucchini-nlp) [inline tests/models/qwen3_vl_moe/test_modeling_qwen3_vl_moe.py]: lets use `assertListEqual` to get a better error message when tests fail

**Review by zucchini-nlp — COMMENTED** [review]
nice, can you make sure that 'unittest.skip' is deleted in MoE model, then I will trigger slow tests

File: tests/models/qwen3_vl_moe/test_modeling_qwen3_vl_moe.py (hunk 5 above)
Comment (JJJYmmm) [reply tests/models/qwen3_vl_moe/test_modeling_qwen3_vl_moe.py]: done!

File: tests/models/qwen3_vl/test_processing_qwen3_vl.py (hunk 3 above)
Comment (JJJYmmm) [reply tests/models/qwen3_vl/test_processing_qwen3_vl.py]: fix🫡

File: src/transformers/models/qwen3_vl/modular_qwen3_vl.py (hunk 1 above)
Comment (JJJYmmm) [reply src/transformers/models/qwen3_vl/modular_qwen3_vl.py]: Got it! just remove `@can_return_tuple`🫡

File: tests/models/qwen3_vl_moe/test_modeling_qwen3_vl_moe.py (hunk 4 above)
Comment (JJJYmmm) [reply tests/models/qwen3_vl_moe/test_modeling_qwen3_vl_moe.py:307]: i'm afraid not yet, because the 30A3 weights haven't been released (though they're coming very soon!), we can merge this PR when it's release!

Comment (JJJYmmm) [reply tests/models/qwen3_vl_moe/test_modeling_qwen3_vl_moe.py:307]: I’ll remove the skip and ping you to run the slow tests once it’s released! 🤗

**Comment by zucchini-nlp** [comment]
run-slow: qwen3_vl, qwen3_vl_moe

**Comment by zucchini-nlp** [comment]
run-slow: qwen3_vl, qwen3_vl_moe

File: tests/models/qwen3_vl_moe/test_modeling_qwen3_vl_moe.py (hunk 4 above)
Comment (zucchini-nlp) [reply tests/models/qwen3_vl_moe/test_modeling_qwen3_vl_moe.py:307]: oh oke, we can fix it after release then

**Review by zucchini-nlp — APPROVED** [review]
Lgtm, we can merge and unskip the slow test after release
